google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ExplainTest.java

```diff
@@ -0,0 +1,317 @@
+
+          Assert.assertEquals(1, row.getColumnCount());
+          assertNotNull(row.getString("QUERY PLAN"));
+          String expectedQueryPlan = "some-plan-node : { subquery_cluster_node : 1 }";
+          Assert.assertEquals(expectedQueryPlan, row.getString("QUERY PLAN"));
+        }
+      }
+    }
+
+    List<ExecuteSqlRequest> requests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
+    assertEquals(1, requests.size());
+    ExecuteSqlRequest request = requests.get(0);
+    assertEquals(EXPLAIN_STATEMENT_QUERY.getSql(), request.getSql());
+    assertEquals(QueryMode.PLAN, request.getQueryMode());
+  }
+
+  public void testExplainAnalyze(String statement) {
+    mockSpanner.clearRequests();
+    final Statement explainAnalyseStatement = Statement.of(statement);
+    try (Connection connection = createConnection()) {
+
+      try (ResultSet resultSet = connection.execute(explainAnalyseStatement).getResultSet()) {
+        int count = 0;
+
+        while (resultSet.next()) {
+          if (count == 1) {
+            Assert.assertEquals(true, false);
+          }
+
+          ++count;
+
+          com.google.cloud.spanner.Struct row = resultSet.getCurrentRowAsStruct();
+
+          Assert.assertEquals(2, row.getColumnCount());
+
+          assertNotNull(row.getString("QUERY PLAN"));
+          String expectedQueryPlan = "some-plan-node : { subquery_cluster_node : 1 }";
+          Assert.assertEquals(expectedQueryPlan, row.getString("QUERY PLAN"));
+
+          assertNotNull(row.getString("EXECUTION STATS"));
+          String expectedExecutionStats = "cpu_time : { unit : msec , total_time : 10 }";
+          Assert.assertEquals(expectedExecutionStats, row.getString("EXECUTION STATS"));
+        }
+      }
+    }
+
+    List<ExecuteSqlRequest> requests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
+    assertEquals(1, requests.size());
+    ExecuteSqlRequest request = requests.get(0);
+    assertEquals(EXPLAIN_STATEMENT_QUERY.getSql(), request.getSql());
+    assertEquals(QueryMode.PROFILE, request.getQueryMode());
+  }
+
+  @Test
+  public void testValidExplain() {
+    String statement = "Explain " + EXPLAIN_STATEMENT_QUERY;
+    testExplain(statement);
+
+    statement = "explain " + EXPLAIN_STATEMENT_QUERY;
+    testExplain(statement);
+
+    statement = "explain     " + EXPLAIN_STATEMENT_QUERY;
+    testExplain(statement);
+
+    statement = "explain \t (" + EXPLAIN_STATEMENT_QUERY + ") ";
+    testExplain(statement);
+
+    statement = "    explain \t ( \n   " + EXPLAIN_STATEMENT_QUERY + "   ) ";
+    testExplain(statement);
+
+    statement = "    ExpLAin  (    " + EXPLAIN_STATEMENT_QUERY + "   ) ";
+    testExplain(statement);
+
+    statement = "    EXPLAIN  (    " + EXPLAIN_STATEMENT_QUERY + "   ) ";
+    testExplain(statement);
+  }
+
+  @Test
+  public void testValidExplainWithFalseAnalyse() {
+    String statement = "    explain (analyse false)      " + EXPLAIN_STATEMENT_QUERY;
```

Comment:
Here and elsewhere: PostgreSQL uses `analyze` instead of `analyse` in all documentation. I know both work, but I think it would be better if we also stuck to `analyze` (except for tests that verify that `analyse` also works).

Reply:
done